Tasks/SonarQubePostTest/PRCA/PostComments-Server.ps1

```diff
@@ -0,0 +1,241 @@
+        "Microsoft.VisualStudio.Services.WebApi.dll")
+       
+    $externalAssemblyPaths = $externalAssemblyNames | foreach { [System.IO.Path]::Combine($tfsClientAssemblyDir, $_)}                        
+    $externalAssemblyPaths | foreach {Add-Type -Path $_} 
+    
+    Write-Verbose "Loaded $externalAssemblyPaths"
+}
+
+
+#
+# Returns a list of files that have been changed in this PR
+# 
+# Remark: public for test purposes
+function GetModifiedFilesInPR 
+{
+    Write-Verbose "Computing the list of files changed in this PR"
+    $sourceFiles = @()
+
+    $targetVersionDescriptor = New-Object -TypeName "Microsoft.TeamFoundation.SourceControl.WebApi.GitTargetVersionDescriptor"
+    $targetVersionDescriptor.VersionType = [Microsoft.TeamFoundation.SourceControl.WebApi.GitVersionType]::Commit
+    $targetVersionDescriptor.Version = $script:pullRequest.LastMergeSourceCommit.CommitId
+    
+    $baseVersionDescriptor = New-Object -TypeName "Microsoft.TeamFoundation.SourceControl.WebApi.GitBaseVersionDescriptor"
+    $baseVersionDescriptor.VersionType = [Microsoft.TeamFoundation.SourceControl.WebApi.GitVersionType]::Commit
+    $baseVersionDescriptor.Version = $script:pullRequest.LastMergeTargetCommit.CommitId
+    
+    $commitDiffs = $script:gitClient.GetCommitDiffsAsync(
+        $script:project, # string project 
+        $script:pullRequest.Repository.Name, # string repositoryId 
+        $true, # bool? diffCommonCommit
+        $null, # int? top 
+        $null, # int? skip
+        $baseVersionDescriptor, 
+        $targetVersionDescriptor, 
+        $null, # object userState
+        [System.Threading.CancellationToken]::None # CancellationToken cancellationToken
+        ).Result
+    
+    if ($commitDiffs.ChangeCounts.Count -gt 0)
+    {
+        Write-Verbose "Found $($commitDiffs.ChangeCounts.Count) changed file(s) in the PR"
+        
+        $sourceFiles = $commitDiffs.Changes | 
+            Where-Object { ($_ -ne $null) -and ($_.Item.IsFolder -eq $false) }  | 
+            ForEach-Object { $_.Item.Path.Replace("\", "/") }
+    }
+
+    return $sourceFiles
+}
+
+#
+# Posts the discussion threads loaded with comments to the PR
+#
+function PostDiscussionThreads
+{
+    param ([ValidateNotNull()][Microsoft.VisualStudio.Services.CodeReview.Discussion.WebApi.DiscussionThreadCollection]$threads)
+    
+    $vssJsonThreadCollection = New-Object -TypeName "Microsoft.VisualStudio.Services.WebApi.VssJsonCollectionWrapper[Microsoft.VisualStudio.Services.CodeReview.Discussion.WebApi.DiscussionThreadCollection]" -ArgumentList @(,$threads)
+    [void]$script:discussionClient.CreateThreadsAsync($vssJsonThreadCollection, $null, [System.Threading.CancellationToken]::None).Result
+    
+    Write-Host "Posted $($threads.Count) discussion threads"
+}
+
+#
+# Retrieve existing discussion threads that were created by this module and that are active (i.e. not fixed)
+#
+function FetchActiveDiscussionThreads
```

Comment:
I made a change here because I found a bug in the original implementation. I am now getting only the "Active" threads - we shouldn't use the "resolved" threads in the logic.